### ISSUE TITLE
Document exposing the gateway on a different port

### ISSUE THREAD

**Issue by LucasRoesler** [openfaas/docs#136]
## Proposed change

The docs should show developers how to change the public port of the gateway. In both swarm and k8s this is not very complicated.  We should warn users about exposing it publically, but for developers testing OpenFaaS they might have port conflicts on their local systems.

The docs should identify use cases and which files would need to be modified. 

## Context

Several users have asked about this in Slack.  Documenting this will simplify the developer experience. 
* Operating System and version (e.g. Linux, Windows, MacOS):

* Link to your project or a code example to reproduce issue:

* Please also follow the [troubleshooting guide](https://github.com/openfaas/faas/blob/master/guide/troubleshooting.md) and paste in any other diagnostic information you have:

**Comment by alexellis** [comment]
Want to `/add label: help wanted` or assign?

**Comment by alexellis** [comment]
> In both swarm and k8s this is not very complicated. 

I don't agree, this is very complicated for Kubernetes and I've seen had anyone ask for that target platform.

I'd see this as only covering Swarm and docker-compose.yml.

**Comment by alexellis** [comment]
@LucasRoesler can you go ahead and pick this up between now and KubeCon, but only for Swarm?

If we start to see users how to change their ingress controllers, then I would be happy to cover that at a later date.
